docs/src/internals/index.md

```diff
@@ -0,0 +1,36 @@
+# Mimi Internals
+
+## Overview
+
+This document provides a forward-looking description of the internal architecture of Mimi. That is, we describe here where Mimi is going, not where it has been.
+
+Several improvements are planned, relative to the prototype (Julia) version of Mimi. These include:
+
+1. Simplification of internal structure by overloading the "." operator
+2. Simplification of the @defcomp macro to improve maintainability
+3. Support for [Monte Carlo](montecarlo.html) simulation
+4. A simple graphical user interface for viewing model results as plots
+5. Renaming of internal structures and types for clarity and consistency
+6. other stuff?
+
+## Renaming of internal structure / types
+We are retaining the distinction between (i) descriptive objects that define the structure (and default values) for models and their components, and (ii) model run-time objects that represent actual model and component instances (i.e., numerous instances of which might be running on a cluster)
+
+* Descriptive object names end with ```Def```, as in ```ModelDef``` and ```ComponentDef```.
+* Run-time objects are the same names without the ```Def```, i.e., ```Model``` and ```Component```.
+
+### Name changes
+* ```Model``` becomes ```ModelDef```
+* ```ModelInstance``` becomes ```Model```
```

Comment:
So this is the one case where I'm not yet sure... I could see a version where we have ``ModelDef``, ``ModelInstacne`` and ``Model``. We can discuss today in the meeting.